tests/test_client_request.py

```diff
@@ -1230,3 +1230,61 @@ def test_loose_cookies_types(loop: Any) -> None:
 def test_gen_default_accept_encoding(has_brotli: Any, expected: Any) -> None:
     with mock.patch("aiohttp.client_reqrep.HAS_BROTLI", has_brotli):
         assert _gen_default_accept_encoding() == expected
+
+
+@pytest.mark.parametrize(
+    ["netrc_contents", "hostname", "trust_env", "expected_auth"],
+    [
+        (
+            "machine example.com login username password pass\n",
+            "example.com",
+            True,
+            helpers.BasicAuth("username", "pass"),
+        ),
+        (
+            "machine example.com login username password pass\n",
+            "example.com",
+            False,
+            None,
+        ),
+    ],
+    indirect=("netrc_contents",),
+)
+@pytest.mark.usefixtures("netrc_contents")
+def test_basicauth_from_netrc_present(
+    make_request: Any,
+    netrc_contents: str,
+    hostname: str,
+    trust_env: bool,
+    expected_auth: Optional[helpers.BasicAuth],
+):
+    """
+    Test appropriate Authorization header is sent when netrc is not empty.
+
+    No authorization header should be sent when trust_env is false, even if
+    netrc is not empty.
+    """
+    req = make_request("get", f"http://{hostname}", trust_env=trust_env)
+    if trust_env:
+        assert req.headers[hdrs.AUTHORIZATION] == expected_auth.encode()
+
+
+@pytest.mark.parametrize(
+    ["netrc_contents", "hostname", "trust_env", "expected_auth"],
+    [
+        ("", "example.com", True, None),
+        ("", "example.com", False, None),
+    ],
+    indirect=("netrc_contents",),
+)
+@pytest.mark.usefixtures("netrc_contents")
+def test_basicauth_from_netrc_absent(
```

Comment:
So the file isn't missing, but empty? Why not just say that?

Reply:
Renamed test case